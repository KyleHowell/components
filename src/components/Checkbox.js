import React from "react";
import PropTypes from "prop-types";
import styled from "react-emotion";

const Wrapper = styled.div`
  cursor: pointer;
  width: 20px;
  height: 20px;
  text-align: center;
  background-color: ${props => (props.checked ? "green" : "red")};
  border-radius: 25%;
`;

const Checkbox = ({ checked, onChange }) => (
  <Wrapper
    checked={checked}
    className="checkbox-component"
    onClick={onChange}
  />
);

Checkbox.propTypes = {
  checked: PropTypes.bool,
  onChange: PropTypes.func
};

export default Checkbox;
