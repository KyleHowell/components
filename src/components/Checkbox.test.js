import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import * as emotion from "emotion";
import { createSerializer, createMatchers } from "jest-emotion";

import Checkbox from "./Checkbox";
import { Message } from "./Checkbox.styles";

expect.addSnapshotSerializer(createSerializer(emotion));
expect.extend(createMatchers(emotion));

Enzyme.configure({ adapter: new Adapter() });

describe("Checkbox Component", () => {
  it("renders correctly when checked", () => {
    const tree = shallow(<Checkbox checked />);
    expect(tree).toMatchSnapshot();
  });
  it("renders correctly when unchecked", () => {
    const tree = shallow(<Checkbox checked={false} />);
    expect(tree).toMatchSnapshot();
  });
  it("renders a green message when checked", () => {
    const tree = shallow(<Message on />);
    expect(tree).toHaveStyleRule("background-color", "green");
  });
  it("renders a red message when unchecked", () => {
    const tree = shallow(<Message on={false} />);
    expect(tree).toHaveStyleRule("background-color", "red");
  });
});
