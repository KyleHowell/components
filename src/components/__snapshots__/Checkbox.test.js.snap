// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox Component renders correctly when checked 1`] = `
<Styled(div)
  className="checkbox-component"
>
  <input
    checked={true}
    type="checkbox"
  />
  <Styled(div)
    on={true}
  >
    On
  </Styled(div)>
</Styled(div)>
`;

exports[`Checkbox Component renders correctly when unchecked 1`] = `
<Styled(div)
  className="checkbox-component"
>
  <input
    checked={false}
    type="checkbox"
  />
  <Styled(div)
    on={false}
  >
    Off
  </Styled(div)>
</Styled(div)>
`;
